pkg/api/norman/customization/cluster/actionhandler.go

```diff
@@ -39,29 +38,37 @@ type ActionHandler struct {
 	Auth                          requests.Authenticator
 }
 
-func (a ActionHandler) ClusterActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
-	canUpdateCluster := func() bool {
-		cluster := map[string]interface{}{
-			"id": apiContext.ID,
-		}
-		return apiContext.AccessControl.CanDo(v3.ClusterGroupVersionKind.Group, v3.ClusterResource.Name, "update", apiContext, cluster, apiContext.Schema) == nil
+func canUpdateCluster(apiContext *types.APIContext) bool {
+	if apiContext == nil {
+		return false
+	}
+	cluster := map[string]interface{}{
+		"id": apiContext.ID,
 	}
+	return apiContext.AccessControl.CanDo(v3.ClusterGroupVersionKind.Group, v3.ClusterResource.Name, "update", apiContext, cluster, apiContext.Schema) == nil
+}
 
-	canBackupEtcd := func() bool {
-		etcdBackupSchema := types.Schema{ID: mgmtclient.EtcdBackupType}
-		// pkg/rbac/access_control.go:55 canAccess checks for the object's ID or namespace. The ns for etcdbackup will be the clusterID
-		backupMap := map[string]interface{}{
-			"namespaceId": apiContext.ID,
-		}
-		return apiContext.AccessControl.CanDo(v3.EtcdBackupGroupVersionKind.Group, v3.EtcdBackupResource.Name, "create", apiContext, backupMap, &etcdBackupSchema) == nil
+func canBackupEtcd(apiContext *types.APIContext) bool {
+	if apiContext == nil {
+		return false
+	}
+	etcdBackupSchema := types.Schema{ID: mgmtclient.EtcdBackupType}
+	backupMap := map[string]interface{} {
+		"namespaceId": apiContext.ID,
 	}
+	return apiContext.AccessControl.CanDo(v3.EtcdBackupGroupVersionKind.Group, v3.EtcdBackupResource.Name, "create", apiContext, backupMap, &etcdBackupSchema) == nil
+}
 
-	canCreateClusterTemplate := func() bool {
-		callerID := apiContext.Request.Header.Get(gaccess.ImpersonateUserHeader)
-		canCreateTemplates, _ := CanCreateRKETemplate(callerID, a.SubjectAccessReviewClient)
-		return canCreateTemplates
+func canCreateClusterTemplate(a ActionHandler, apiContext *types.APIContext) bool {
```

Comment:
I think it would be better to make this a method on the `ActionHandler`:
```suggestion
func (a ActionHandler) canCreateClusterTemplate(apiContext *types.APIContext) bool {
```